.github/workflows/ci.yml

```diff
@@ -1,47 +1,82 @@
+on:
+  push:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - closed
 
 jobs:
   test:
     runs-on: ubuntu-latest
 
     steps:
-    - name: Checkout code
-      uses: actions/checkout@v4
-
-    - name: Set up Python
-      uses: actions/setup-python@v5
-      with:
-        python-version: 3.11
-
-    - name: Install Poetry
-      run: |
-        python -m pip install --upgrade pip
-        python -m pip install poetry
-
-    - name: Install dependencies
-      working-directory: ./space2stats_api/src
-      run: |
-        poetry install --with test
-
-    - name: install lib postgres
-      uses: nyurik/action-setup-postgis@v2
-
-    - name: Run pre-commit
-      working-directory: ./space2stats_api/src
-      run: |
-        poetry run pre-commit run --all-files
-
-    - name: Run tests
-      working-directory: ./space2stats_api/src
-      run: |
-        poetry run python -m pytest --benchmark-skip tests
-      env:
-        PGHOST: localhost
-        PGPORT: 5432
-        PGDATABASE: mydatabase
-        PGUSER: myuser
-        PGPASSWORD: mypassword
-        PGTABLENAME: space2stats
-        S3_BUCKET_NAME: test-bucket
\ No newline at end of file
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install Poetry
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install poetry
+
+      - name: Install dependencies
+        working-directory: ./space2stats_api/src
+        run: |
+          poetry install --with test
+
+      - name: install lib postgres
+        uses: nyurik/action-setup-postgis@v2
+
+      - name: Run pre-commit
+        working-directory: ./space2stats_api/src
+        run: |
+          poetry run pre-commit run --all-files
+
+      - name: Run tests
+        working-directory: ./space2stats_api/src
+        run: |
+          poetry run python -m pytest --benchmark-skip tests
+        env:
+          PGHOST: localhost
+          PGPORT: 5432
+          PGDATABASE: mydatabase
+          PGUSER: myuser
+          PGPASSWORD: mypassword
+          PGTABLENAME: space2stats
+          S3_BUCKET_NAME: test-bucket
+
+  deploy-to-dev:
+    if: ${{ github.event_name == 'pull_request' }}
+    uses: "./.github/workflows/deploy.yml"
+    needs: test
+    with:
+      environment: Space2Stats API Dev
+      stage: pr-${{ github.event.pull_request.number }}
+      pr-number: ${{ github.event.pull_request.number }}
+    secrets: inherit
+
+  deploy-to-production:
+    if: ${{ github.event_name == 'push' && github.ref_name == 'main' }}
+    uses: "./.github/workflows/deploy.yml"
+    needs: test
+    with:
+      environment: Space2Stats API Prod
+      stage: prod
+    secrets: inherit
+
+  destroy-pr-preview:
+    if: ${{ github.event.action == 'closed' }}
+    uses: "./.github/workflows/deploy.yml"
```

Comment:
Should this not be calling destroy?

Reply:
Yup, caught in [324c3cb](https://github.com/worldbank/DECAT_Space2Stats/pull/78/commits/324c3cb7cabac9c9119287766a5f17e606ebf258) and I am refining in #78.  Feel free to have a 👀